prow/tide/search.go

```diff
@@ -27,121 +27,50 @@ import (
-				break
-			}
-			vars["searchCursor"] = githubql.NewString(sq.Search.PageInfo.EndCursor)
-		}
-		log.WithFields(logrus.Fields{
-			"query":    datedQuery,
-			"start":    start.String(),
-			"end":      end.String(),
-			"duration": time.Since(requestStart).String(),
-		}).Debugf("Query returned %d PRs and cost %d point(s). %d remaining.", len(ret), totalCost, remaining)
-		return ret, totalMatches, nil
-	}
-}
-
-func (q searchExecutor) search() ([]PullRequest, error) {
-	prs, _, err := q.searchRange(time.Time{}, time.Now())
-	return prs, err
-}
-
-func (q searchExecutor) searchSince(t time.Time) ([]PullRequest, error) {
-	prs, _, err := q.searchRange(t, time.Now())
-	return prs, err
-}
-
-func (q searchExecutor) searchRange(start, end time.Time) ([]PullRequest, int, error) {
-	// Adjust times to be after GitHub was founded to avoid querying empty time
-	// ranges.
+func search(query querier, log *logrus.Entry, q string, start, end time.Time) ([]PullRequest, error) {
 	if start.Before(github.FoundingYear) {
 		start = github.FoundingYear
 	}
 	if end.Before(github.FoundingYear) {
 		end = github.FoundingYear
 	}
-
-	prs, count, err := q(start, end)
-	if err != nil {
-		return nil, 0, err
-	}
-
-	if count <= 1000 {
-		// The search returned all the results for the query.
-		return prs, len(prs), nil
-	}
-	// The query returned too many results, we need to partition it.
-	prs, err = q.partitionSearchRange(start, end, count)
-	return prs, len(prs), err
-}
-
-func (q searchExecutor) partitionSearchRange(start, end time.Time, count int) ([]PullRequest, error) {
-	partition := partitionTime(start, end, count, 900)
-	// Search right side...
-	rPRs, rCount, err := q.searchRange(partition, end)
-	if err != nil {
-		return nil, err
-	}
-
-	// Search left side...
-	// For the left side we can deduce the count in advance.
-	lCount := count - rCount
-	// If the count is too large we can skip the initial search and go straight to
-	// partitioning to save an API token.
-	var lPRs []PullRequest
-	if lCount <= 1000 {
-		lPRs, _, err = q.searchRange(start, partition)
-	} else {
-		lPRs, err = q.partitionSearchRange(start, partition, lCount)
-	}
-	if err != nil {
-		return nil, err
+	log = log.WithFields(logrus.Fields{
+		"query": q,
+		"start": start.String(),
+		"end":   end.String(),
+	})
+	requestStart := time.Now()
+	datedQuery := fmt.Sprintf("%s %s", q, dateToken(start, end))
+	var cursor *githubql.String
+	vars := map[string]interface{}{
+		"query":        githubql.String(datedQuery),
+		"searchCursor": cursor,
 	}
-
-	return append(lPRs, rPRs...), nil
-}
-
-func partitionTime(start, end time.Time, count, goalSize int) time.Time {
-	duration := end.Sub(start)
-	if count < goalSize*2 {
-		// Choose the midpoint.
-		return start.Add(duration / 2)
+	var totalCost, remaining int
+	var ret []PullRequest
+	var sq searchQuery
+	for {
+		log.Debug("Sending query")
+		if err := query(context.Background(), &sq, vars); err != nil {
+			return ret, err
```

Comment:
nit: This error message won't be very actionable without knowing what query it is associated with. What is the motivation for changing this?

Reply:
`q` is passed into this function and invariant. Therefore we do not need to include it in the error message we return. The caller to `search()` can decide how they want to handle the returned error.

The caller either logs an error message that includes the query, or else returns an error with the query included.